### PR TITLE
feat: Migration soak test

### PR DESCRIPTION
**Pull request by petretiandrea**
This PR initialize the repo with soak test suite. It contains a setup with typescript to write k6 performance/soak test.

Actually only migration soak test are implemented

#### List of Changes

- added `soak-test` folder
- setup typescript project with k6 like eCommerce-tests
- added migration soak tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,4 +99,17 @@ Template parameters:
...
+Every step of migration test will generate a new contract id by calling PUT /migrations/wallets and then
+will update or delete the wallet using POST /migrations/updateDetails or POST /migrations/delete.
+
+| Parameter key                | Type   | Description                                             |
+|------------------------------|--------|---------------------------------------------------------|
+| API_SUBSCRIPTION_KEY                  | string | API key used to finalize contract import    |
+| URL_BASE_PATH                  | string | URL base path to import contract api group    |
+| API_SUBSCRIPTION_KEY_GENERATE_CONTRACT                  | string | API key used by generating contract     |
+| URL_BASE_PATH_GENERATE_CONTRACT                  | string | URL base path to generate contract api group    |
+| DELETE_RATIO  | number | Between 0 and 1. The percentage of contract DELETE  |
```
Comment (giovanniberti) [inline README.md:115]: table formatting

```suggestion
| Parameter key                          | Type   | Description                                        |
|----------------------------------------|--------|----------------------------------------------------|
| API_SUBSCRIPTION_KEY                   | string | API key used to finalize contract import           |
| URL_BASE_PATH                          | string | URL base path to import contract api group         |
| API_SUBSCRIPTION_KEY_GENERATE_CONTRACT | string | API key used by generating contract                |
| URL_BASE_PATH_GENERATE_CONTRACT        | string | URL base path to generate contract api group       |
| DELETE_RATIO                           | number | Between 0 and 1. The percentage of contract DELETE |
```

File: soak-test/src/common/config.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export type K6Config = t.TypeOf<typeof K6Config>;
+
+export type IConfig = t.TypeOf<typeof IConfig>;
+export const IConfig = t.intersection([
+  t.interface({
+    URL_BASE_PATH: NonEmptyString,
+    API_SUBSCRIPTION_KEY: NonEmptyString,
+    API_SUBSCRIPTION_KEY_GENERATE_CONTRACT: NonEmptyString,
+    URL_BASE_PATH_GENERATE_CONTRACT: NonEmptyString,
+    DELETE_RATIO: NumberFromString,
```
Comment (giovanniberti) [inline soak-test/src/common/config.ts:25]: Maybe add a check that this is between 0 and 1?

File: .devops/azure-templates/soak-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+parameters:
```
Comment (giovanniberti) [inline .devops/azure-templates/soak-tests.yaml:1]: Missing `DELETE_RATIO` configuration?

**Review by giovanniberti — APPROVED** [review]

File: .devops/azure-templates/soak-tests.yaml (hunk 3 above)
Comment (petretiandrea) [reply .devops/azure-templates/soak-tests.yaml:1]: Solved by https://github.com/pagopa/pagopa-wallet-tests/pull/33/commits/da855ea0c5f593a13180c5c0544d779fe4729246

File: soak-test/src/common/config.ts (hunk 2 above)
Comment (petretiandrea) [reply soak-test/src/common/config.ts]: Solved by https://github.com/pagopa/pagopa-wallet-tests/pull/33/commits/da855ea0c5f593a13180c5c0544d779fe4729246

**Pull request merged**